lib/jarvis/commands/merge.rb

```diff
@@ -72,40 +72,42 @@ def execute
     Jarvis::Git.config(git, "user.email", committer_email)
     Jarvis::Git.config(git, "user.name", committer_name)
 
+    if branches.empty?
+      logger.info("No branches given, merging to PR's base branch: #{pull.base.ref}")
+      target_branch = pull.base.ref
+      backport_branches = []
+    else
+      target_branch, *backport_branches = branches
+
+      if pull.base.ref != target_branch
+        raise "First branch in the list doesn't match the PR's target branch. Expected '#{pull.base.ref}', got '#{target_branch}'"
+      end
+    end
+
     commits = []
-    branches.each do |branch|
+
+    logger[:operation] = "git merge"
+    logger[:branch] = target_branch
+    pr_head = git.revparse("HEAD")
+    git.checkout(target_branch)
+    git.merge(pr_head)
+    logger.info("Merged to target branch: \"#{target_branch}\"")
+
+    backport_branches.each do |branch|
       logger[:branch] = branch
-      logger.info("Working on a branch")
+      logger.info("Backport to branch: \"#{branch}\"")
       git.checkout(branch)
 
       commits << { :branch => branch, :commits => [] }
       patches = Mbox.new(File.new(patch_file, "r"))
       patches.each do |mail|
-        ::Jarvis::Git.apply_mail(git, mail, logger) do |description|
-          # Append the 'Fixes #1234' to the bottom of the commit message for
-          # whatever PR this is..
-          description + "\nFixes \##{pr.number}"
-        end
-
-        # Verify the committer is set correctly.
-        data = git.lib.commit_data("HEAD")
-        # format of the committer field is this: Your Name <foo@example.com> 1234567890 -0800
-        _, name, email, *_ = /^([^<]+) <([^>]+)> \d+ [+-]\d+/.match(data["committer"]).to_a
-        raise Bug, "Committer name didn't match: #{name} vs #{committer_name}" if name != committer_name
-        raise Bug, "Committer email didn't match: #{email} vs #{committer_email}" if email != committer_email
-
-        # Record the commit hash for this commit.
-        commits.last[:commits] << git.revparse("HEAD") 
+        ::Jarvis::Git.apply_mail(git, mail, logger)
+        commits.last[:commits] << git.revparse("HEAD")
```

Comment:
Since we discussed adding meta-data to the issue anyway, why not just add the 'Closes: #PR-Number' so that we have that meta-data available in the commit message anyway?

Reply:
I tried to explain this here:

> The new merge command will not append a "fixes #pr_number" sentence to
the commit message as it did before. Doing this would create a different
commit preventing the github PR from being marked as merged.

Mutating the original commit means the PR won't be marked "merged". if we don't mind the merged vs closed PR status then we can keep the previous behaviour and I'll just replace  the fixes with closes